File: program/find-the-average-of-all-numbers-in-an-array/find_the_average_of_all_numbers_in_an_array.go
package main

import "fmt"

func main() {
	array := []int{1, 2, 5, 3, 6, 4, 7, 8, 9, 10}
	n := len(array)
	sum := 0
	for i := 0; i < n; i++ {
		sum += (array[i])
	}
	avg := (float64(sum)) / (float64(n))
	fmt.Println(avg)
}

File: program/program/find-the-average-of-all-numbers-in-an-array/FindTheAverageOfAllNumbersInAnArray.fs
let avg aList =
    let rec sum = function
        | head :: tail -> head + (sum tail)
        | [] -> 0.
    sum aList / (aList |> List.length |> float)

let res = avg [ 1.; 2.; 3.; 4.; 5. ]
printfn "Average : %A" res
